src/euphorie/client/browser/risk.py

```diff
@@ -509,6 +509,61 @@ def action_plan_condition(self):
             return None
         return condition
 
+    def save_data(self, reply):
+        old_values = {}
+        for prop, default in self.monitored_properties.items():
+            val = getattr(self.context, prop, default)
+            old_values[prop] = val
+
+        self.set_answer_data(reply)
+
+        session = Session()
+        changed = self.set_measure_data(reply, session)
+
+        if reply.get("answer", None):
+            # If answer is not present in the request, do not attempt to set
+            # any action-related data, since the request might have come
+            # from a sub-form.
+            if self.webhelpers.integrated_action_plan:
+                new_plans, changes = self.extract_plans_from_request()
+                for plan in (
+                    self.context.standard_measures + self.context.custom_measures
+                ):
+                    session.delete(plan)
+                self.context.action_plans.extend(new_plans)
+                changed = changes or changed
+
+        # This only happens on custom risks
+        if reply.get("handle_custom_description"):
+            self.context.custom_description = self.webhelpers.check_markup(
+                reply.get("custom_description")
+            )
+
+        if reply.get("title"):
+            self.context.title = reply.get("title")
+
+        if not changed:
+            for prop, default in self.monitored_properties.items():
+                val = getattr(self.context, prop, None)
+                if val and val != old_values[prop]:
+                    changed = True
+                    break
+        if changed:
+            self.session.touch()
+
+    def post(self):
+        reply = self.request.form
+        if not self.webhelpers.can_edit_session:
+            return self.proceed_to_next(reply)
+        _next = self._get_next(reply)
+        # Don't persist anything if the user skipped the question
+        if _next == "skip":
+            return self.proceed_to_next(reply)
+
+        self.save_data(reply)
+
+        return self.proceed_to_next(reply)
```

Comment:
This would untangle the code a bit for the cost of not being a simple restructuring anymore.
Just ignore this.

```suggestion
        if self.webhelpers.can_edit_session:
            _next = self._get_next(reply)
            # Only persist, if the user hasn't skipped the question
            if _next != "skip":
                self.save_data(reply)

        return self.proceed_to_next(reply)
```

Reply:
I did consider this, but went for only splitting up the methods for now. Maybe I'll get around to it while working on the DS etool.